### PR TITLE
Use ssl.match_hostname from urllib3 as it was removed from Python 3.12

### PR DESCRIPTION
**Pull request by rcritten**
See https://pagure.io/freeipa/issue/9409
and https://github.com/python/cpython/pull/94224#issuecomment-1621097418

### CODE REVIEW

**Comment by rcritten** [comment]
So urllib3.util.ssl_match_hostname was introduced in version 1.26.8. RHEL 9.x has 1.26.5 and I can't tell what version Azure has. I'll add a conditional import.

Full release notes https://github.com/urllib3/urllib3/blob/main/CHANGES.rst

**Comment by frasertweedale** [comment]
python3-urllib3 dependency needs to be added to `python3-ipalib`  Requires in spec file.

Code changes LGTM.

File: freeipa.spec.in [hunk 1]
```diff
@@ -896,6 +896,12 @@ Requires: platform-python-setuptools
 %else
 Requires: python3-setuptools
 %endif
+# Indirect dependency: use newer urllib3 with TLS 1.3 PHA support
```
[the remark this answers is not in the document]
Comment (rcritten) [reply freeipa.spec.in:899]: This is just a copy of the existing placeholder for now while I fight with Azure to install the right dependencies in tox.

**Comment by abbra** [comment]
LGTM.

**Comment by flo-renaud** [comment]
master:

* d2ed490ff446d96520b89ea47387ce8ee33c1c7d Use ssl.match_hostname from urllib3 as it was removed from Python 3.12

**Pull request closed without merging**